### ISSUE TITLE
0.46 provides post transaction handlers, could potentially migrate MsgFeeHandler to that 

### ISSUE THREAD

**Issue by arnabmitra** [provenance-io/provenance#1018]
## Summary

0.46 provides post transaction handlers, could potentially migrate MsgFeeHandler to that.

## Problem Definition
```
PostHandler are like AnteHandler (they share the same signature), but they execute after [RunMsgs](https://docs.cosmos.network/master/core/baseapp.html#runmsgs).

Like AnteHandlers, PostHandlers are theoretically optional, 

```
Also 
```
Note, when PostHandlers fail, the state from runMsgs is also reverted, effectively making the transaction fail.
```
So fit's well, may require some finagling on current events being emitted but probably possible?

This would eliminate changes in our cosmos fork, make code easier to maintain etc.

## Proposal
1.Add posthandler to provenance.
```
func NewPostHandler(options HandlerOptions) (sdk.AnteHandler, error) {
	// pseudo code
	postDecorators := []sdk.AnteDecorator{
		NewMsgFeeDecorator(options.MsgFeeKeeper),
	}

	return sdk.ChainAnteDecorators(postDecorators...), nil
}
```
2.Migrate MsgFeeHandler to be a postHandler
```
func (d msgFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error)
```
3. verify wasm message's work and other tests pass.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by SpicyLemon** [comment]
They changed a bunch of stuff around with v0.46 and we need to fix it before we can ship v1.12. #1006 is for doing that.

They also replaced `NewMempoolDecorator` with `NewDeductFeeDecorator`. The `NewDeductFeeDecorator` function takes in a `TxFeeChecker` (`func(ctx sdk.Context, tx sdk.Tx) (sdk.Coins, int64, error)`) that might also be leveraged. So there's options.

**Comment by SpicyLemon** [comment]
I'm going to close this as a duplicate of #1006. There's some good stuff in here that will help with that one though.
